functions/_pure_place_iterm2_prompt_mark.fish

```diff
@@ -0,0 +1,5 @@
+function _pure_place_iterm2_prompt_mark
+    if functions -q iterm2_prompt_mark
+        iterm2_prompt_mark
```

Comment:
Here is iTerm integration script: https://raw.githubusercontent.com/gnachman/iterm2-website/master/source/shell_integration/fish

iTerm places a special mark on the line containing invoked command. I need to fix the mark placement, so it will be on the second line of the prompt (since this is where user input is located). To do that I invoke `iterm2_prompt_mark` if it exists (if the user uses iTerm with shell integration, it will be defined) before printing the second line of the prompt.

iTerm integration script checks `fish_prompt` for existence of `iterm2_prompt_mark` string. If it exists, it assumes, that the user (in that case it's `pure` prompt) placed mark manually, so integration script doesn't need to place it itself. This is why the function needs to be inside `fish_prompt.fish` and this is why it's called `_pure_place_`**`iterm2_prompt_mark`**.

Reply:
I wonder if there is a more elegant solution to this, but so far that's the best I came up with.